omniscidb/QueryBuilder/QueryBuilder.h

```diff
@@ -0,0 +1,656 @@
+/**
+ * Copyright (C) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "IR/Exception.h"
+#include "IR/Node.h"
+#include "SchemaMgr/SchemaProvider.h"
+#include "Shared/Config.h"
+
+namespace hdk::ir {
+
+class InvalidQueryError : public Error {
+ public:
+  InvalidQueryError() {}
+  InvalidQueryError(std::string desc) : Error(std::move(desc)) {}
+};
+
+class QueryBuilder;
+class BuilderNode;
+class ExprRewriter;
+
+class BuilderExpr {
+ public:
+  BuilderExpr(const BuilderExpr& other) = default;
+  BuilderExpr(BuilderExpr&& other) = default;
+
+  BuilderExpr& operator=(const BuilderExpr& other) = default;
+  BuilderExpr& operator=(BuilderExpr&& other) = default;
+
+  BuilderExpr name(const std::string& name) const;
+  const std::string& name() const { return name_; }
```

Comment:
These two:
```
  BuilderExpr name(const std::string& name) const;
  const std::string& name() const { return name_; }
```
look like a getter/setter pair at a first glance, might benefit from renaming.

Reply:
How about `BuilderExpr rename(const std::string& name) const`? E. g. `node.ref("value").sum().rename("total_value")`.

Reply:
Yeah, I was thinking of `rename` or `named`. I like `rename`.

Reply:
Done